index.js

```diff
@@ -696,11 +726,20 @@ answer.encode = function (a, buf, offset) {
 
   buf.writeUInt16BE(types.toType(a.type), offset)
 
-  let klass = classes.toClass(a.class === undefined ? 'IN' : a.class)
-  if (a.flush) klass |= FLUSH_MASK // the 1st bit of the class is the flush bit
-  buf.writeUInt16BE(klass, offset + 2)
-
-  buf.writeUInt32BE(a.ttl || 0, offset + 4)
+  if (a.type === 'OPT') {
+    if (a.name !== '.') {
+      throw new Error('OPT name must be root.')
+    }
+    buf.writeUInt16BE(a.udp || 4096, offset + 2)
+    buf.writeUInt8(a.extRcode || 0, offset + 4)
+    buf.writeUInt8(a.ednsVersion || 0, offset + 5)
+    buf.writeUInt16BE(0, offset + 6)
```

Comment:
suggest: `buf.writeUInt16BE(a.flags || 0, offset + 6)` to allow DO bit to be set.

Reply:
If we're going to use `class` as an alias for `udpPayloadSize`, we should allow `ttl` as an alias here.

Reply:
Hmm, I think I'd prefer this to be further split up, e.g. `flag_do` and `flag_z` properties.

Reply:
Also, I now think aliases would be a bad idea, better be strict about it.